compiler/rustc_lint/src/context/diagnostics.rs

```diff
@@ -171,6 +171,9 @@ pub(super) fn decorate_lint(sess: &Session, diagnostic: BuiltinLintDiag, diag: &
             }
             .decorate_lint(diag);
         }
+        BuiltinLintDiag::ReservedString(space_span) => {
+            lints::ReservedString { suggestion: space_span }.decorate_lint(diag);
```

Comment:
This can be written like this
```suggestion
        BuiltinLintDiag::ReservedString(suggestion) => {
            lints::ReservedString { suggestion }.decorate_lint(diag);
```